Popover API - add live examples for Web API (#26893)
mdn/content: files/en-us/web/api/htmlbuttonelement/popovertargetaction/index.md

## Examples

> ### Toggle popover action

Author: ```suggestion
### Toggle popover action with an auto popover
```

> This example shows the basic use of the popover API with a "toggle" value set for the `popoverTargetAction` property.

Author: The popover is created with the `auto` value, so it can be closed ("light-dismissed") by clicking outside the popover area.

> This example shows the basic use of the popover API with a "toggle" value set for the `popoverTargetAction` property.

Author: ```suggestion
This example shows the basic use of the popover API with a "toggle" value set for the `popoverTargetAction` property.
The popover is created with the `auto` value, so it can be closed ("light-dismissed") by clicking outside the popover area.
```

> First we define an HTML `button` element that we will use to control display and hide the popover, and a `div` that will be the popover.

Author: ```suggestion
First we define an HTML `<button>` element that we will use to show and hide the popover, and a `<div>` that will be the popover.
```

In this case we don't set the [`popovertargetaction`](/en-US/docs/Web/HTML/Element/button#popovertargetaction) HTML attribute on the button or the [`popover`](/en-US/docs/Web/HTML/Global_attributes/popover) attribute on the `div`, as we will be doing so programmatically.

```html
<button id="toggleBtn">Toggle popover</button>
<div id="mypopover">This is popover content!</div>
```

The JavaScript code first gets a handle to the div element and the button.

> It then defines a function to feature check for popover support.

Author: ```suggestion
It then defines a function to check for popover support.
```

```js
const popover = document.getElementById("mypopover");
const toggleBtn = document.getElementById("toggleBtn");

> // Feature check for popover API support.

Author: ```suggestion
// Check for popover API support.
```

function supportsPopover() {
  return HTMLElement.prototype.hasOwnProperty("popover");
}
```

If the popover API is supported the code sets the `div` element's `popover` attribute to `"auto"` and makes it the popover target of the toggle button.
We then set the `popoverTargetAction` of the button to `"toggle"`.

> If the popover API is not supported we simply change the text content of the `div` element to state this, and hide the toggle button.

Author: Only if we tell him :-). It's a good point though.

```suggestion
If the popover API is not supported we change the text content of the `<div>` element to state this, and hide the toggle button.
```

```js
if (supportsPopover()) {
  // Set the div element to be an auto popover
  popover.popover = "auto";
  // Set the button popover target to be the popover
  toggleBtn.popoverTargetElement = popover;

  // Set that the button toggles popover visibility
  toggleBtn.popoverTargetAction = "toggle";
} else {
  popover.textContent = "Popover API not supported.";
  toggleBtn.hidden = true;
}
```

> **Note:** A popover element is hidden by default, but if the API is not supported your element will display "as usual".

You can try out the example below.
Show and hide the popover by toggling the button.
The "auto" popover can also be dismissed by selecting outside the bounds of the popover text.

> {{EmbedLiveSample("Toggle popover action", "100%")}}

Author: ```suggestion
{{EmbedLiveSample("Toggle popover action with an auto popover", "100%")}}
```

> ### Show/hide popover action

Author: ```suggestion
### Show/hide popover action with a manual popover
```

> This example shows how to use the `"show"` and `"hide"` values of the `popoverTargetAction` attribute.

Author: ```suggestion
This example shows how to use the `"show"` and `"hide"` values of the `popoverTargetAction` attribute.
The `popover` attribute is set to `manual`, so the popover must be closed using a button, and not by selecting outside the popover area.
```

The code is near identical to the previous example, except that there are two buttons, and the popover is set to "manual".

```html
<button id="showBtn">Show popover</button>
<button id="hideBtn">Hide popover</button>
<div id="mypopover">This is popover content!</div>
```

```js
function supportsPopover() {
  return HTMLElement.prototype.hasOwnProperty("popover");
}

const popover = document.getElementById("mypopover");
const showBtn = document.getElementById("showBtn");
const hideBtn = document.getElementById("hideBtn");

const popoverSupported = supportsPopover();

if (supportsPopover()) {
  // Set the div element be a manual popover
  popover.popover = "manual";

  // Set the button targets to be the popover
  showBtn.popoverTargetElement = popover;
  hideBtn.popoverTargetElement = popover;

  // Set the target actions to be show/hide
  showBtn.popoverTargetAction = "show";
  hideBtn.popoverTargetAction = "hide";
} else {
  popover.textContent = "Popover API not supported.";
  showBtn.hidden = true;
  hideBtn.hidden = true;
}
```

The popover can be displayed by selecting the "Show popover" button, and dismissed using the "Hide popover" button.

> Note that the popover has been set to `"manual"`, so it can't be dismissed by selecting outside the popover.

Author: ```suggestion
Note that the popover has been set to `"manual"`, so it can't be light dismissed by selecting outside the popover.
```

> {{EmbedLiveSample("Show/hide popover action", "100%")}}

Author: ```suggestion
{{EmbedLiveSample("Show/hide popover action with a manual popover", "100%")}}
```

Show and hide the popover by toggling the button.
The `"auto"` popover can also be "light dismissed" by selecting outside the bounds of the popover text.

{{EmbedLiveSample("Toggle popover action with an auto popover", "100%")}}

### Show/hide popover action with a manual popover

This example shows how to use the `"show"` and `"hide"` values of the `popoverTargetAction` attribute.

The code is near identical to the previous example, except that there are two `<button>` elements, and the popover is set to [`"manual"`](/en-US/docs/Web/API/Popover_API/Using#using_manual_popover_state).
A `manual` popover must be closed explicitly, and not "light dismissed" by selecting outside the popover area.

```html
<button id="showBtn">Show popover</button>
<button id="hideBtn">Hide popover</button>
<div id="mypopover">This is popover content!</div>
```

```js
function supportsPopover() {
  return HTMLElement.prototype.hasOwnProperty("popover");
}

const popover = document.getElementById("mypopover");
const showBtn = document.getElementById("showBtn");
const hideBtn = document.getElementById("hideBtn");

const popoverSupported = supportsPopover();

if (supportsPopover()) {

>   // Set the div element be a manual popover

Author: ```suggestion
  // Set the <div> element be a manual popover
```
